### DISCUSSION TITLE
xrdp + x11vnc: UTF-8 encoding is treated as ISO-8859-1 in clipboard transfers

### DISCUSSION THREAD

**Discussion by usa-** [neutrinolabs/xrdp#3207] [Q&A]
I use x11vnc as VNC server and UTF-8 as the locale charset.
I use Remote Desktop client to connect to xrdp, which has the following connection settings:
```
[Xvnc]
name=Xvnc
lib=libvnc.so
user=ask
password=ask
ip=127.0.0.1
port=5900
```

- When I copy "Привет, мир!!!" at the client and paste it at server, I get ", !!".
- When I copy "Привет, мир!!!" at the server and paste it at client, I get "Ð�Ñ�Ð¸Ð²ÐµÑ�, Ð¼Ð¸Ñ�!!!".
- When I copy this gibberish at the client and paste it back at the server, I get "Привет, мир!!!" back.
- When I type any unicode text and save it in a file with UTF-8 encoding, open this file as if it is in ISO-8859-1 encoding, copy its contents at the client and paste at the server, I get the correct initial unicode text.

**Comment by matt335672** [comment]
The basic RFB protocol specified ISO-8859-1 as the character set for cut-and-paste. There's an extension for UTF-8, but we don't support it at this level.

You can run `xrdp-chansrv` in your desktop session. If you do this, you can specify `chansrvport=DISPLAY(0)` in `xrdp.ini`. You'll then not only get UTF-8 cut-and-paste, but you'll be able to paste images, access files, etc.

**Reply by matt335672** [reply]
PS : you'll need xrdp v0.9.16 of later to do this.

**Reply by usa-** [reply]
Works with v0.9.26.
For some reason, does not work with the latest release (v.0.10.1) and development version v.0.10.80. Just nothing happens when I copy-paste in any direction. xrdp-chansrv does not report any error.
